app/services/email_output_service.rb

```diff
@@ -0,0 +1,70 @@
+class EmailOutputService
+  def initialize(email_service:)
+    @email_service = email_service
+  end
+
+  def execute(submission_id:, action:, attachments:, pdf_attachment:)
+    email_attachments = []
+
+    if action.fetch(:include_attachments) == true
+      email_attachments = email_attachments.concat attachments
+    end
+    if action.fetch(:include_pdf) == true
+      email_attachments << pdf_attachment
+    end
+
+    if email_attachments.empty?
+      send_single_email(
+        action: action,
+        attachments: [],
+        subject: subject(
+          subject: action.fetch(:subject),
+          current_email: 1,
+          number_of_emails: 1,
+          submission_id: submission_id
+        )
+      )
+    else
+      send_emails_with_attachments(action, email_attachments, submission_id: submission_id)
+    end
```

Comment:
This `execute` method is pretty long, I'm looking at it and wondering if we can do something like:
```
def execute
  attachments = gather_attachments
  send_email(attachments)
end
```
This sums up the service a bit better. But looks like it may be tricky with all the arguments. More something for you to consider and change if you think its possible.